app/src/main/java/org/oppia/android/app/preview/TopicPreviewActivity.kt

```diff
@@ -0,0 +1,52 @@
+package org.oppia.android.app.preview
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import org.oppia.android.app.activity.InjectableAppCompatActivity
+import javax.inject.Inject
+
+/** TopicPreviewFragment TAG. */
+const val TOPIC_PREVIEW_FRAGMENT_TAG = "TopicPreviewFragment"
+private const val TOPIC_PREVIEW_ACTIVITY_TOPIC_ID_ARGUMENT_KEY = "TopicPreviewActivity.topic_id"
+private const val TOPIC_PREVIEW_ACTIVITY_PROFILE_ID_ARGUMENT_KEY = "TopicPreviewActivity.profile_id"
+
+/** Activity display preview of topic when topic is not downloaded. */
+class TopicPreviewActivity : InjectableAppCompatActivity() {
+
+  companion object {
+    /**
+     * This function creates the intent of [TopicPreviewActivity].
+     *
+     * @param [internalProfileId]: Id of the Profile.
+     * @param [topicId]: Id of the Topic.
+     *
+     * @return [Intent]: TopicPreviewActivity Intent object.
```

Comment:
```suggestion
     * @return an [Intent] to open the [TopicPreviewActivity]
```

We do not explicitly specify the type in the return at-clause (it's not needed since KDoc generation will pull the type from the method declaration itself).

Ditto elsewhere.